src/napari/_vispy/canvas.py

```diff
@@ -387,6 +389,19 @@ def delete(self) -> None:
         """Schedules the native widget for deletion"""
         self.native.deleteLater()
 
+    @contextlib.contextmanager
+    def quitting(self):
+        """Context manager to signal that napari is quitting.
+
+        The `_is_quitting` attribute will be used by _remove_layer to
+        skip expensive operations that are unneeded during shutdown.
+        """
+        self._is_quitting = True
+        try:
+            yield
+        finally:
+            self._is_quitting = False
```

Comment:
```suggestion
        prev = self._is_quitting
        self._is_quitting = True
        try:
            yield
        finally:
            self._is_quitting = prev
```
Always restore the previous value.